### ISSUE TITLE
[Tech Request]: Improve testing of Query Plans for Master/Vector/Secondary Index

### ISSUE THREAD

**Issue by arjunsk** [matrixorigin/matrixone#15019]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

```Markdown
The performance of Index queries is not tested in regression tests. We need a strategy to test
- Plan generated by Index Queries
- Data stored inside the Index Hidden table
- Performance metrics of Index Queries over different runs
```

### Describe the feature you'd like

Possible approaches to solve this
- Modification to BVT framework to do Plan comparison
- Running a Python code to validate the data stored in the hidden table.
- Running a Python code to validate the QPS of Index queries.

**Comment by arjunsk** [comment]
As suggested by @fengttt , one approach is to support

`EXPLAIN FORMAT=JSON select * from t3;`. Then query the JSON string to do regex checks on whether `BlockFilter` is present or not etc.

**Comment by arjunsk** [comment]
I don't think we support REGEX, LENGTH, etc functions on Query Plan output from "EXPLAIN".

**Comment by arjunsk** [comment]
A. For Vector Index

1. Search Query Correctness BVT is improved by using precise coordinates.
2. Vector Insert Performance will monitored with this test: https://github.com/matrixorigin/matrixone/issues/15018
3. Vector QPS performance needs to be evaluated with this test


B. For Master Index
No plan yet

C. For Secondary Index
No plan yet

**Comment by cpegeric** [comment]
I recommend to use random vector generation with same seek so that we can always generate same set of vectors without storing the data in database.

See the code here,
https://github.com/cpegeric/wiki-benchmark/blob/main/python/indextest.py

The script support the following,
1. build hnsw/ivfflat index with different op_type (vector_ip_ops, vector_cosine_ops, vector_l2_ops), any dimension
2. run recall to check the recall rate.  Our target recall rate is > 0.9

**Comment by heni02** [comment]
下一版本验证

**Comment by heni02** [comment]
目前回归已加入ivf index benchmark测试
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/16753103693
